packages/hd/src/keystore.ts

```diff
@@ -230,18 +217,15 @@ export default class Keystore {
   }
 
   static mac(derivedKey: Buffer, ciphertext: Buffer): HexStringWithoutPrefix {
-    const keccakSize = 256;
-
-    return (
-      new Keccak(keccakSize)
-        // https://github.com/ethereumjs/ethereumjs-wallet/blob/d57582443fbac2b63956e6d5c4193aa8ce925b3d/src/index.ts#L615-L617
-        // eslint-disable-next-line @typescript-eslint/no-magic-numbers
-        .update(Buffer.concat([derivedKey.subarray(16, 32), ciphertext]))
-        .digest("hex")
+    // https://github.com/ethereumjs/ethereumjs-wallet/blob/d57582443fbac2b63956e6d5c4193aa8ce925b3d/src/index.ts#L615-L617
+    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
+    const hash = keccak_256(
+      Buffer.concat([derivedKey.subarray(16, 32), ciphertext])
     );
+    return Buffer.from(hash).toString("hex");
   }
 
-  static scryptOptions(kdfparams: KdfParams): ScryptOptions {
+  static scryptOptions(kdfparams: KdfParams): ScryptOpts {
```

Comment:
Is `ScryptOpts` the same as `ScryptOptions`? It would be better to copy it from "crypto.d.ts" to ensure the method is not changed

Reply:
Both `ScryptOptions` and `ScryptOpts` provide inaccurate return type.
Based on the method body, the mort accurate return type should be just no explicit return type.
```ts
  static scryptOptions(kdfparams: KdfParams) {
    return {
      N: kdfparams.n,
      r: kdfparams.r,
      p: kdfparams.p,
      // eslint-disable-next-line @typescript-eslint/no-magic-numbers
      maxmem: 128 * (kdfparams.n + kdfparams.p + 2) * kdfparams.r,
    };
  }
```

Reply:
I'm not sure I understand the meaning of "inaccurate return type" 

The original method returns the type of:

```ts
interface InferReturnType {
      N: number,
      r: number,
      p: number,
      maxmem: number,
}
```

This type is compatible with the definition `ScryptOptions` in `crypto.d.ts`. So, what does "inaccurate type" refer to?

Changing `ScryptOptions` to `ScryptOpts` is a breaking change that could lead to unexpected results for users who rely on the API directly.

Reply:
I simply removed the return type of this method in latest update.

This is how the method looks like now.
```ts
  static scryptOptions(kdfparams: KdfParams) {
    return {
      N: kdfparams.n,
      r: kdfparams.r,
      p: kdfparams.p,
      // eslint-disable-next-line @typescript-eslint/no-magic-numbers
      maxmem: 128 * (kdfparams.n + kdfparams.p + 2) * kdfparams.r,
    };
  }
```


This is the accurate return type
```ts
interface InferReturnType {
      N: number,
      r: number,
      p: number,
      maxmem: number,
}
```
Both `ScryptOptions` and `ScryptOpts` are just OK, but inaccurate.

Reply:
Or can you define the return type explicitly?

```diff
+type ScryptOptions = {
+  N: number;
+  r: number;
+  p: number;
+  maxmem: number;
+}
```

Also, remove the `ScryptOptions` from import, as it is compatible with both `ScryptOpts` and `crypto.ScryptOptions`

Reply:
Done.